crates/shared/src/encoded_settlement.rs

```diff
@@ -84,6 +85,64 @@ pub struct EncodedSettlement {
     pub interactions: [Vec<EncodedInteraction>; 3],
 }
 
+impl EncodedSettlement {
+    /// Order uids for all trades in the settlement.
```

Comment:
Nit: comment here and on the `trades()` that the `Vec` will retain the order of trades as they are executed on chain (since you rely on this detail in the `zip` function)

Reply:
Added comments